exec/pipeline/financials/structs.py

```diff
@@ -0,0 +1,140 @@
+    services: Decimal
+    total_net_sales: Decimal
+
+class CostofSales(BaseModel):
+    products: Decimal
+    services: Decimal
+    total_cost_of_sales: Decimal
+
+
+
+class OperatingExpenses(BaseModel):
+    research_and_development: Decimal
+    selling_general_and_administrative: Decimal
+    total__operating_expenses: Decimal
+
+
+class CurrentAssets(BaseModel):
+    cash_and_cash_equivalents: Decimal
+    marketable_securities: Decimal
+    accounts_receivable_net: Decimal
+    vendor_non_trade_receivables: Decimal
+    other_current_assets: Decimal
+    total_current_assets: Decimal
+
+    # @validator("total_current_assets")
+    # def total_current_assets_is_sum_of_all_assets(cls, v):
+    #     return v == (
+    #         cls.cash_and_cash_equivalents
+    #         + cls.marketable_securities
+    #         + cls.accounts_receivable_net
+    #         + cls.vendor_non_trade_receivables
+    #         + cls.other_current_assets
+    #     )
+
+
+
+
+class NonCurrentAssets(BaseModel):
+    marketable_securities: Decimal
+    property_plant_and_equipment_net: Decimal
+    other_non_current_assets: Decimal
+    total_non_current_assets: Decimal
+
+
+class Assets(BaseModel):
+    current_assets: CurrentAssets
+    non_current_assets: NonCurrentAssets
+
+
+class EarningsPerShare(BaseModel):
+    basic: Decimal
+    diluted: Decimal
+
+class Shares(BaseModel):
+    basic: int
+    diluted: int
+
+
+
+class NetSalesByReportableSegment(BaseModel):
+    americas: Decimal
+    europe: Decimal
+    greater_china: Decimal
+    japan: Decimal
+    rest_of_asia_pacific: Decimal
+    total_net_sales: Decimal
+
+
+class NetSalesByCategory(BaseModel):
+    iphone: Decimal
+    mac: Decimal
+    ipad: Decimal
+    wearable_home_and_accessories: Decimal
+    services: Decimal
+    total_net_sales: Decimal
+
+
+class CurrentLiabilities(BaseModel):
+    accounts_payable: Decimal
+    other_current_liabilities: Decimal
+    defeerred_revenue: Decimal
+    commercial_paper: Decimal
+    term_debt: Decimal
+    total_current_liabilities: Decimal
+
+
+class NonCurrentLiabilities(BaseModel):
+    term_debt: Decimal
+    other_non_current_liabilities: Decimal
+    total_non_current_liabilities: Decimal
+
+
+
+class Liabilities(BaseModel):
+    current_liabilities: CurrentLiabilities
+    non_current_liabilities: NonCurrentLiabilities
+    total_liabilities: Decimal
+
+class FinancialStatement(BaseModel):
+    time_window: str
+    net_sales: NetSales
+    cost_of_sales: CostofSales
+    gross_margin: Decimal
+    operating_expenses: OperatingExpenses
+    operating_income: Decimal
+    other_income_net: Decimal
+    income_before_provision_for_taxes: Decimal
+    provision_for_taxes: Decimal
+    net_income: Decimal
+    earnings_per_share: EarningsPerShare
+    shares_used_in_computing_earnings_per_share: Shares
+    assets: Assets
+    liabilities: Liabilities
+    commitmements_and_contingencies: Decimal
+
+
+
+class EarningsStatement(BaseModel):
```

Comment:
What is this class used for? Genuinely curious.

Reply:
this is passed into the OpenAI client to tell it how to parse the output, that way we get structured outputs from the LLM we can use as data